src/tsconfig.ts

```diff
@@ -41,7 +41,12 @@ const recommended: Record<string, any> = {
 }
 
 const build: Record<string, any> = {
-  extends: '../tsconfig.json',
+  extends:
+    config.project === undefined
+      ? '../tsconfig.json'
+      : /* c8 ignore start */
+        join('..', config.project),
+  /* c8 ignore stop */
```

Comment:
cc @isaacs -- are you ok with this? The behavior is being tested via the integration test (`basic-custom-project`). Unclear whether there's a good way to test it at unit test level.

Reply:
This should be covered by the tests in test/tsconfig.ts